### ISSUE TITLE
VC++ Complains About strdup Function

### ISSUE THREAD

**Issue by ChrisDrozdowski** [steven-varga/h5cpp#19]
Compiling with VC++ in Visual Studio 2017, the compiler complains about strdup usage:

`Error	C4996	'strdup': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _strdup.`

It is referring to line 88 in H5Dwrite.hpp:
```
try {
	for( const auto& reference:ref)
		ptr.push_back( strdup( reference.data()) );
} catch( ... ){
	throw h5::error::io::dataset::write( h5::error::msg::mem_alloc );
}
```

**Comment by steven-varga** [comment]
Hi Chris,
checkout the windows branch and send me a pull request please.

**Comment by ChrisDrozdowski** [comment]
Hi,

I already added a fix for this in commit: fb7f77c66878f5ec44f6141d23e9443d386c9e73
